src/tracing/mod.rs

```diff
@@ -43,27 +45,26 @@ enum TracingResult {
 
 impl TracingResult {
     /// Converts the tracing result into Geth traces.
-    fn into_geth(self) -> Option<Vec<TraceResult>> {
-        if let Self::Geth(traces) = self {
-            Some(traces)
+    fn as_geth(&self) -> Option<Vec<TraceResult>> {
+        if let Self::Geth(ref traces) = self {
+            Some(traces.clone())
         } else {
             None
         }
     }
 
     /// Converts the tracing result into Parity traces.
-    fn into_parity(self) -> Option<Vec<LocalizedTransactionTrace>> {
-        if let Self::Parity(traces) = self {
-            Some(traces)
+    fn as_parity(&self) -> Option<Vec<LocalizedTransactionTrace>> {
+        if let Self::Parity(ref traces) = self {
+            Some(traces.clone())
```

Comment:
do we need to clone? I think we can return a `&Vec<LocalizzedTransactionTrace>` no?

Reply:
true just fixed and rearrange the place where it is used to satisfy the requirements